packages/nextjs/src/config/webpack.ts

```diff
@@ -214,3 +199,56 @@ function checkWebpackPluginOverrides(userSentryWebpackPluginOptions: Partial<Sen
     );
   }
 }
+
+/**
+ * Determine if this is an entry point into which both `Sentry.init()` code and the release value should be injected
+ *
+ * @param entryPointName The name of the entry point in question
+ * @returns `true` if sentry code should be injected, and `false` otherwise
+ */
+function shouldAddSentryToEntryPoint(entryPointName: string): boolean {
+  return entryPointName === 'pages/_app' || entryPointName.includes('pages/api');
+}
+
+/**
+ * Combine default and user-provided SentryWebpackPlugin options, accounting for whether we're building server files or
+ * client files.
+ *
+ * @param buildContext Nexjs-provided data about the current build
+ * @param userPluginOptions User-provided SentryWebpackPlugin options
+ * @returns Final set of combined options
+ */
+function getWebpackPluginOptions(
+  buildContext: BuildContext,
+  userPluginOptions: Partial<SentryWebpackPluginOptions>,
+): SentryWebpackPluginOptions {
+  const { isServer, dir: projectDir, buildId, dev: isDev } = buildContext;
+
+  const hasSentryProperties = fs.existsSync(path.resolve(projectDir, 'sentry.properties'));
+
+  const serverInclude = [
+    { path: '.next/server/chunks/', urlPrefix: '~/_next/server/chunks' },
+    { path: '.next/server/pages/', urlPrefix: '~/_next/server/pages' },
+    { path: '.next/serverless/', urlPrefix: '~/_next/serverless' },
+  ];
```

Comment:
This feels brittle - are we sure that we are not missing any other includes here?

Reply:
Dead sure? No. Pretty sure? Yes. The alternative is listing the many things we want to ignore, and this felt cleaner. Definitely open to a third option if you have a different idea.

Reply:
> Definitely open to a third option if you have a different idea.

I spent a bit thinking about this, and I couldn't get anywhere so lets just go with this for now.